src/watts/plugin_dakota.py

```diff
@@ -0,0 +1,268 @@
+
+        super().prerun(params)
+        if 'dakota_driver_name' in params.keys():
+            os.chmod(params['dakota_driver_name'], 0o755)
+
+    @property
+    def execute_command(self):
+        return [str(self.executable), "-i", self.input_name]
+
+
+def run_dakota_driver(coupled_code_exec: str):
+    """ Function to execute the workflow for data
+    exchange between Dakota and the coupled code
+    in the Dakota driver script.
+
+    Parameters
+    ----------
+    coupled_code_exec
+        The name of the WATTS python script of the
+        coupled code.
+    """
+    results = parse_dakota_input()
+    retval = run_coupled_code(coupled_code_exec)
+    return_dakota_input(results, retval)
+
+def parse_dakota_input():
+    """Parse Dakota input
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    results
+        Results from Dakota's output file
+    """
+    
+    from interfacing import interfacing as di # Dakota's interface module        
+
+    params, results = di.read_parameters_file()
+
+    # Dump params to external params.json file for future use by the template engine
+    params_for_template_engine_file_path = "params.json"
+    with open(params_for_template_engine_file_path, 'w') as outfile:
+        f = json.dump(params._variables,  outfile, default=lambda o: o.__dict__)
+    return(results)
+
+def run_coupled_code(coupled_code_exec):
+    """ Run the coupled code
+
+    Parameters
+    ----------
+    coupled_code_exec
+        The name of the WATTS python script of the
+        coupled code.
+
+    Returns
+    -------
+    retval
+        Processed output from the coupled code.
+    """
+
+    if os.path.exists(coupled_code_exec):
+        P = subprocess.check_output(["python", coupled_code_exec])
+    else:
+        raise RuntimeError("Coupled-code script missing.")
+
+    # Read the 'opt_res.out' pickle file and 
+    # store the results to 'res_output' for data
+    # transfer with Dakota.
+    if os.path.exists('opt_res.out'): 
+        db = pickle.load(open('opt_res.out', 'rb'))
+        if 'dakota_descriptors' in db.keys():
+            res_output = []
+            for key in db['dakota_descriptors']:
+                res_output.append(db[db['dakota_descriptors'][key]])
+    else:
+        raise RuntimeError("'opt_res.out' file is missing.")
+
+    retval = dict([])
+    retval['fns'] = res_output
+    return(retval)
+
+def return_dakota_input(results, retval):
+    """ Return the output of the coupled code
+    to Dakota.
+
+    Parameters
+    ----------
+    results
+        Results from Dakota's output file
+    retval
+        Processed output from the coupled code.
+    """
+
+    # Insert extracted values into results
+    # Results iterator provides an index, response name, and response
+    try:
+      for i, n, r in results:
+        if r.asv.function:
+            try:
+                r.function = retval['fns'][i]
+            except:
+                pass
```

Comment:
A "catch-all" `except` clause is considered bad practice. If you are expecting a particular exception, it should be listed explicitly.

Reply:
@nstauff I copied this from your previous implementation. Do you know what exceptions are we expecting here?

Reply:
@nstauff and I aren't sure what exceptions are expected here. I couldn't recreate the exceptions because this block of code seems to be applicable only to the earlier versions of Dakota. What would you suggest I do here @paulromano ? Thanks.

Reply:
Ok, I guess we'll leave this as is -- please add a comment about the `try:` block that briefly explains why it is there